fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -97,4 +107,88 @@ public Collection<LoanDelinquencyTagHistoryData> retrieveDelinquencyRangeHistory
         return mapperLoanDelinquencyTagHistory.map(loanDelinquencyTagData);
     }
 
+    @Override
+    public CollectionData calculateLoanCollectionData(final Long loanId) {
+        final LocalDate businessDate = DateUtils.getBusinessLocalDate();
+        final Loan loan = this.loanRepository.getReferenceById(loanId);
+
+        CollectionData collectionData = CollectionData.template();
+        collectionData.setAvailableDisbursementAmount(loan.getApprovedPrincipal().subtract(loan.getDisbursedAmount()));
+        collectionData.setNextPaymentDueDate(loan.possibleNextRepaymentDate());
+
+        final MonetaryCurrency currency = loan.getCurrency();
+        BigDecimal delinquentAmount = BigDecimal.ZERO;
+        // Overdue Days calculation
+        Long overdueDays = 0L;
+        LocalDate overdueSinceDate = null;
+        final List<LoanTransaction> chargebackTransactions = loan.retreiveListOfTransactionsByType(LoanTransactionType.CHARGEBACK);
+        if (chargebackTransactions.isEmpty()) {
+            for (LoanTransaction loanTransaction : chargebackTransactions) {
+                Set<LoanTransactionToRepaymentScheduleMapping> loanTransactionToRepaymentScheduleMappings = loanTransaction
+                        .getLoanTransactionToRepaymentScheduleMappings();
+                LoanTransactionToRepaymentScheduleMapping loanTransactionToRepaymentScheduleMapping = loanTransactionToRepaymentScheduleMappings
+                        .iterator().next();
+                if (!loanTransactionToRepaymentScheduleMapping.getLoanRepaymentScheduleInstallment().isPrincipalCompleted(currency)) {
+                    overdueSinceDate = loanTransaction.getTransactionDate();
+                    break;
+                }
+            }
+        }
+
+        final List<LoanRepaymentScheduleInstallment> installments = loan.getRepaymentScheduleInstallments();
+        if (overdueSinceDate == null) {
+            for (LoanRepaymentScheduleInstallment installment : installments) {
+                if (installment.isNotFullyPaidOff()) {
+                    overdueSinceDate = installment.getDueDate();
+                    break;
+                }
+            }
+        }
+
+        if (overdueSinceDate != null) {
+            if (loan.loanProduct().getLoanProductConfigurableAttributes().getGraceOnArrearsAgingBoolean()) {
+                final Integer graceDays = loan.loanProduct().getLoanProductRelatedDetail().getGraceOnArrearsAgeing();
+                if (graceDays != null) {
+                    overdueSinceDate = overdueSinceDate.plusDays(graceDays);
+                }
+            }
+            overdueDays = DateUtils.getDifferenceInDays(businessDate, overdueSinceDate);
+            if (overdueDays < 0) {
+                overdueDays = 0L;
+            }
+            collectionData.setDelinquentDate(overdueSinceDate);
+        }
+
+        collectionData.setPastDueDays(overdueDays.intValue());
```

Comment:
Are you sure the "past due days" and "delinquent days" are equals? If so, why do we have 2 field with same value?

Reply:
Yes, It was the original implementation

<img width="1443" alt="Screenshot 2022-10-26 at 17 08 06" src="https://user-images.githubusercontent.com/44206706/198155124-9ceedbe8-b41e-45d8-a6ac-5d877e8a34d5.png">